### DISCUSSION TITLE
"Switch session"/"Close current session and open this session" option

### DISCUSSION THREAD

**Discussion by dot3dash4dot** [sienori/Tab-Session-Manager#822] [Ideas]
I've started using my laptop for both work and personal purposes, and that's led me to a new workflow that's a bit "clicky" at the moment. Basically, I want to be able to finish my work each evening and do the following:

1. Replace my saved "Work" session with the session I have open
2. Close the current session
3. Open my saved "Home" session

In order to do this at the moment, step 1 is pretty self-explanatory, but steps 2 involve opening the saved session and then manually going through the windows I have open to close the "Work" ones that are still open. Or alternatively, closing all but one of the "Work" windows, using that last one to launch the "Home" session, and then closing the remaining "Work" window.

Is there a quicker way of doing this that I've missed? If not, my proposal is for each session's menu to have a "Switch session" option which closes the current session and opens the selected session instead. Or if that's not explanatory enough, it could be called "Close current session and open this session". This would make steps 2 and 3 above a simple 3-click flow rather than having to manually sort through windows.

**Comment by sienori** [comment]
Use "Open in current window" to close other windows and open the session.
![image](https://user-images.githubusercontent.com/29514424/127770273-11d6b9d2-60e2-4ffd-9712-8bffe0598c0c.png)
